backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/nodes/MergedReferencesCompartmentItemNodeDescriptionProvider.java

```diff
@@ -0,0 +1,126 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.syson.diagram.common.view.nodes;
+
+import java.util.List;
+import java.util.Objects;
+
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.emf.ecore.EClassifier;
+import org.eclipse.emf.ecore.EReference;
+import org.eclipse.emf.ecore.ETypedElement;
+import org.eclipse.sirius.components.view.builder.providers.IColorProvider;
+import org.eclipse.sirius.components.view.diagram.InsideLabelDescription;
+import org.eclipse.sirius.components.view.diagram.InsideLabelPosition;
+import org.eclipse.sirius.components.view.diagram.InsideLabelStyle;
+import org.eclipse.sirius.components.view.diagram.NodeDescription;
+import org.eclipse.sirius.components.view.diagram.NodePalette;
+import org.eclipse.sirius.components.view.diagram.NodeStyleDescription;
+import org.eclipse.sirius.components.view.diagram.SynchronizationPolicy;
+import org.eclipse.syson.util.AQLUtils;
+import org.eclipse.syson.util.IDescriptionNameGenerator;
+import org.eclipse.syson.util.SysMLMetamodelHelper;
+import org.eclipse.syson.util.ViewConstants;
+
+/**
+ * Used to create the Compartment item node description.
+ *
+ * @author adieumegard
+ */
+public class MergedReferencesCompartmentItemNodeDescriptionProvider extends AbstractNodeDescriptionProvider {
+
+    private final EClass eClass;
+
+    private final List<EReference> eReferences;
+
+    private final EClassifier eType;
+
+    private final IDescriptionNameGenerator descriptionNameGenerator;
+
+    public MergedReferencesCompartmentItemNodeDescriptionProvider(EClass eClass, List<EReference> eReferences, IColorProvider colorProvider, IDescriptionNameGenerator nameGenerator) {
+        super(colorProvider);
+        this.eClass = Objects.requireNonNull(eClass);
+        this.eReferences = Objects.requireNonNull(eReferences);
+        if (this.eReferences.stream().map(ETypedElement::getEType).distinct().count() != 1) {
+            Objects.requireNonNull(null);
```

Comment:
Please remove this or fix it

Reply:
In the previous version it was raising an "IllegalArgumentException" but a test fails if we introduce a new XXXException (org.eclipse.syson.tests.GeneralPurposeTests.testNoThrowNewException(int, String, Path, List<String>)). So I decided to default with the standard checks (which "throw new NullPointerException").
Just wandering if the test is not a bit strong.
Searching for an alternative...

Reply:
What's the problem if several references have the same EType?

Reply:
Good question ! I was assuming all references shall have the same EType but after all why not.